src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs

```diff
@@ -143,7 +143,7 @@ public static partial class ImmutableArray
         public void CopyTo(T[] destination) { }
         public void CopyTo(T[] destination, int destinationIndex) { }
         public bool Equals(System.Collections.Immutable.ImmutableArray<T> other) { throw null; }
-        public override bool Equals(object? obj) { throw null; }
+        public override bool Equals([System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] object? obj) { throw null; }
```

Comment:
This is not how the particular attribute is meant to be used. It is intended for aiding static analysis of `out` parameter nullability.

Reply:
This is continuation of #47598, but if this kind of annotation is no longer relevant, you can close PR

Reply:
Ah interesting, I had not realized the attribute could be used that way. I would say that it is by all means relevant.